app/models/geocode_logic.rb

```diff
@@ -0,0 +1,118 @@
+    elsif geocoded1.present?
+      update_mismatch(result, geocoded1)
+    elsif geocoded2.present?
+      update_mismatch(result, geocoded2)
+    else
+      check_bad_address(result, geocoded3)
+    end
+  end
+
+  def check_bad_address(result, geocoded3)
+    # uses geocoder text search and looks for coords by Institution name if applicable
+    int = result.institution
+    geocode_name = Geocoder.search(int.downcase.split('#').first.downcase) if int.present?
+    geocode_name.present? ? full_text_search(result, geocode_name, geocoded3) : search_bad_addr(result, geocoded3)
+  end
+
+  def parse_add_fields(res, field)
+    string = "#{field}, #{res.city}, #{res.state}, #{res.zip}, #{res.country}"
+    return string if field.present?
+  end
+
+  def parse_address(res, field)
+    field.present? ? "#{field}, #{res.city}, #{res.physical_country}" : "#{res.city}, #{res.physical_country}"
+  end
+
+  def search_bad_addr(result, geocoded3)
+    # parses bad addresses and flags and issues
+    stopwords = %w[ave avenue road rd dr drive blvd pkwy st street parkway]
+    stopwords_regex = /\b(#{Regexp.union(*stopwords).source})\b/i
+    add_split = result.address.split(stopwords_regex).map(&:strip)[0..1].join(' ')
+    new_address = [add_split, result.state, result.zip].compact.join(' ')
+    geocoded = Geocoder.coordinates(new_address)
+    update_bad_addr(result, geocoded, geocoded3)
+  end
+
+  def update_bad_addr(result, geocoded, geocoded3)
+    if geocoded.present?
+      update_mismatch(result, geocoded)
+    else
+      update_mismatch(result, geocoded3)
+      result.update(bad_address: true)
+    end
+  end
+
+  def full_text_search(result, geocode_name, geocoded3)
+    check_search_fields = []
+    geocode_name.each do |geo|
+      search_fields = check_srch_fields(result, geo)
+      check_search_fields << search_fields if search_fields.present?
+      next if search_fields.blank?
+
+      upd_data = [search_fields.first.data['lat'].to_f, search_fields.first.data['lon'].to_f]
+      update_mismatch(result, upd_data)
+    end
+    search_bad_addr(result, geocoded3) if check_search_fields.empty?
+  end
+
+  def check_srch_fields(result, geo)
+    state = geo.data['address']['ISO3166-2-lvl4'].present? ? geo.data['address']['ISO3166-2-lvl4'].split('-').last : nil
+    city = geo.data['address']['city']
+    num = geo.data['address']['amenity'].present? ? geo.data['address']['amenity'].split(' ').last.to_i : nil
+    if (num.is_a? Numeric) && (num > 1)
+      return_num_text_search(result, geo, city, state, num)
+    else
+      return_text_search(result, geo, city, state, num)
+    end
+  end
+
+  def return_num_text_search(result, geo, city, state, num)
+    text_search = []
+    add_num = result.institution.downcase.split('#').last.to_i
+    text_search << geo if city == result.city.titleize && state == result.state && num == add_num
+    text_search
+  end
+
+  def return_text_search(result, geo, city, state, _num)
```

Comment:
```suggestion
  def text_search(result, geo, city, state)
```

Reply:
Try not to start method names with `return` for readability. Changing the name to `text_search` will still imply that it returns the same result.
 
Is `num` needed here? It's not ever used inside of the method.